<?xml version="1.0"?>

<launch>

	<arg name="robot_name" default="thorvald_001"/>
	<arg name="tf_prefix" default="$(arg robot_name)/"/>

	<group ns="$(arg robot_name)">
		<!-- launch robot_pose_publisher -->
		<node pkg="robot_pose_publisher" type="robot_pose_publisher" name="robot_pose_publisher" respawn="true">
			<param name="~map_frame" value="map"/>
			<param name="~base_frame" value="$(arg tf_prefix)base_link"/>
		</node>
		
		<!-- move_base -->
		<node name="move_base" pkg="move_base" type="move_base" output="screen">
	        <remap from="odom" to="odometry/base_raw" />
	        <remap from="cmd_vel" to="nav_vel" />
			<remap from="map" to="/map"/>

	        <rosparam file="$(find uol_cmp9767m_tutorial)/config/costmap_common_params.yaml" command="load" ns="global_costmap"/>
	        <rosparam file="$(find uol_cmp9767m_tutorial)/config/costmap_common_params.yaml" command="load" ns="local_costmap"/>
	        <rosparam file="$(find uol_cmp9767m_tutorial)/config/global_costmap_params.yaml" command="load"/>
	        <rosparam file="$(find uol_cmp9767m_tutorial)/config/local_costmap_params.yaml" command="load"/>
	        <rosparam file="$(find uol_cmp9767m_tutorial)/config/planners.yaml" command="load"/>
		</node> 
	</group>

</launch>
